src/features/auth/auth.service.ts

```
@@ -11,17 +13,26 @@ export class AuthService {
 
   async validateUser(username: string, pass: string): Promise<any> {
     const user = await this._usersService.findOne(username);
-    if (user && user.password === pass) {
+    if (user && await bcrypt.compare(pass, user.password)) {
       const { password, ...result } = user;
       return result;
     }
     return null;
```

Instead of returning null, throw a new error